search/query/multi_phrase.go

```diff
@@ -26,9 +26,11 @@ import (
 )
 
 type MultiPhraseQuery struct {
-	Terms    [][]string `json:"terms"`
-	Field    string     `json:"field,omitempty"`
-	BoostVal *Boost     `json:"boost,omitempty"`
+	Terms     [][]string `json:"terms"`
+	Field     string     `json:"field,omitempty"`
+	BoostVal  *Boost     `json:"boost,omitempty"`
+	Fuzziness int        `json:"fuzziness"`
+	Prefix    int        `json:"prefix_length"`
```

Comment:
I'm not sure prefix makes much sense with multi phrase, because the order of terms with multi phrase query does not matter. Also this is more in line with `TermQuery` so let's keep the definition of the MultiPhraseQuery like it was before and apply these new attributes only for a `MatchPhraseQuery`.

Reply:
But a match phrase query / phrase query is internally a multi-phrase query, so it must also be supported.
A prefix essentially is applied to the fuzzy search of each term in the Phrase, so it works out like so, each word will be fuzzily matched to a word in the index, provided that a prefix for the word is present.
The unit test file also reflects the same.

Reply:
basically yeah, lets say i have a match phrase for 
"Apple banana" it becomes a multi phrase query for
[["apple"]["banana"]], and for both of these terms, we have a fuzzy search being done,
So FuzzySearch ("apple", Fuzziness, prefix) is done, so both fuzziness+prefix must be passed from the top level searcher.

If i have MultiPhraseQuery of:
[[ "apple"   ],[ "banana" ] ]
      "red"           "yellow"

Q1 = DisjunctionQuery of FuzzyQueries over "apple" and "red"
Q2 = DisjunctionQuery of FuzzyQueries over "banana" and "yellow"
And result Q = ConjQuery(Q1,Q2). So i have to send prefix via the top searcher or hard code it to 0 while allowing for a user defined fuzziness

Reply:
In Elastic search, Fuzzy Match phrase is not supported out of the box, but can be implemented using some workaround as suggested in this post:
https://stackoverflow.com/questions/38816955/elasticsearch-fuzzy-phrases

Basically use a span query that allows fine-tuning - **for each word the user can give a fuzziness and a prefix length** there as in this image 
<img width="1169" alt="image" src="https://github.com/blevesearch/bleve/assets/53022689/a13c1401-4a70-49d7-9b59-41180b1dc1fe">

The below image shows a match provided a per word fuzziness and prefix
<img width="1306" alt="image" src="https://github.com/blevesearch/bleve/assets/53022689/5d277a95-4741-4a51-be73-ea9779d62ead">


So what i can suggest is - We allow this restricted match phrase query which keeps the fuzziness and prefix same throughout, and then add a functionality to implement span queries for more fine tuning.

Reply:
Done